### PR TITLE
Modal and Card Animation

### LINKED ISSUE #91: Modal and Card Hover Effects
**Is your feature request related to a problem? Please describe.**
Modal opens with no animation.
Implement card hover effect for improved UI.

**Describe the solution you'd like**
When hovering the pokecard a hover animation will be used
When the pokeCard is clicked the modal animates into the window rather than just appearing.

### PR DESCRIPTION
**Pull request by Gjhnsn**
## Changes
1. Implemented React Spring for modal animation
2. Card hover animation
3. Search bar animation
4. Adjusted link hovers

## Purpose
_Describe the problem or feature in addition to a link to the issues._
Polished overall app appearance and interface for an improved user experience.

## Approach
Using React Spring and CSS transitions

https://react-spring.io/

Closes #91

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "react-modal": "^3.14.3",
     "react-scripts": "4.0.3",
     "react-select": "^4.3.1",
+    "react-spring": "^9.2.4",
```
Comment (BlackCubes) [inline package.json:22]: What's dis 👀

File: src/components/PokeCard.js [hunk 2]
```diff
@@ -84,22 +95,24 @@ const PokeCard = ({
...
+            typeTags={typeTags}
+            getPokemonModalAboutContent={getPokemonModalAboutContent}
+            height={height}
+            weight={weight}
+            abilities={abilities}
+            stats={stats}
+            types={types}
+            modalData={modalData}
+          />
+        </animated.div>
```
Comment (dannysanchez559) [inline src/components/PokeCard.js:115]: i see how you implemented that modal animation, looks straightfoward and came out nice!

**Review by BlackCubes — APPROVED** [review]
Everything looks really good!

**Review by dannysanchez559 — APPROVED** [review]
"LGTM"

**Review by connielion — APPROVED** [review]
Looks good to me! Nice job adding react-spring!

**Review by JorgeLVilla — APPROVED** [review]
Awesome job!

**Pull request merged**